feat: add launchctl completion spec.

joeheyming opened the pull request:
This is a basic implementation of launchctl.
We don't want to attempt to list the launchctl labels or service targets.  That can be a future exercise.

**What kind of change does this PR introduce? (Bug fix, feature, docs update, ...)**
Feature

**What is the current behavior? (You can also link to an open issue here)**
Currently autocomplete doesn't support launchctl

**What is the new behavior (if this is a feature change)?**
Now launchctl is supported

**Additional info:**

joeheyming commented:
Please add the `hacktoberfest-accepted` or `hacktoberfest` label to this review so it can count towards my contributions.

grant0417 commented:
Thanks for the contrib! Have been wanting this one, could you use `launchctl list` as a generator for some of the arguments?

joeheyming commented:
Originally, I added the list generator, but it didn't work for all commands.  I felt like it was too much work , so I removed it.  But since you asked, I added it back for commands that use a "label" arg.  But I would have liked to also make it work for service targets and stuff like that, but I think that will have to be future work.

grant0417 commented:
Seems reasonable, LGTM, will let @SeparateRecords merge though!
